sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessorClient.cs

```diff
@@ -35,22 +37,22 @@ public class EventProcessorClient : IAsyncDisposable
         private readonly object StartProcessorGuard = new object();
 
         /// <summary>The handler to be called just before event processing starts for a given partition.</summary>
-        private Func<PartitionContext, Task> _initializeProcessingForPartitionAsync;
+        private Func<InitializePartitionProcessingContext, Task> _initializeProcessingForPartitionAsync;
 
         /// <summary>The handler to be called once event processing stops for a given partition.</summary>
-        private Func<PartitionContext, PartitionProcessorCloseReason, Task> _processingForPartitionStoppedAsync;
+        private Func<PartitionProcessingStoppedContext, Task> _processingForPartitionStoppedAsync;
 
-        /// <summary>Responsible for processing sets of events received from the Event Hubs service.</summary>
-        private Func<PartitionContext, IEnumerable<EventData>, Task> _processEventsAsync;
+        /// <summary>Responsible for processing events received from the Event Hubs service.</summary>
+        private Func<EventProcessorEvent, Task> _processEventAsync;
 
         /// <summary>Responsible for processing unexpected exceptions thrown while this <see cref="EventProcessorClient" /> is running.</summary>
```

Comment:
nit:  All exceptions are unexpected.   :)

Reply:
I think `unhandled` is the word I want.